### PR TITLE
add: blast

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-1337
Figma link (if any):

## What changed (plus any additional context for devs)

added missing params for migration and state migrations

you can check the ticket to repro and double check the fix works

## Screen recordings / screenshots

image

### CODE REVIEW

File: src/core/types/chains.ts [hunk 1]
```diff
@@ -20,6 +21,26 @@ export const chainHardhat: Chain = {
...
+  },
+  blockExplorers: {
+    default: { name: 'Blastscan', url: 'https://blastscan.io/' },
+  },
+  nativeCurrency: {
+    name: 'Blast',
+    symbol: 'BLAST',
+    decimals: 18,
+  },
+};
```
Comment (greg-schrammel) [inline src/core/types/chains.ts:42]: can't we update wagmi/viem? I think they got blast already

File: src/core/resources/assets/userAssets.ts [hunk 2]
```diff
@@ -248,6 +248,7 @@ export async function parseUserAssets({
     (dict, currentChainId) => ({ ...dict, [currentChainId]: {} }),
     {},
   ) as ParsedAssetsDictByChain;
+  console.log('- parsedAssetsDict assets', assets);
```
Comment (greg-schrammel) [inline src/core/resources/assets/userAssets.ts:251]: logs?

File: src/core/types/chains.ts (hunk 1 above)
Comment (dereknelson) [reply src/core/types/chains.ts:42]: i updated it & didn't see it

Comment (DanielSinclair) [reply src/core/types/chains.ts:42]: Should be in `2.7.18` and above

Comment (estebanmino) [reply src/core/types/chains.ts:42]: bumping

Comment (estebanmino) [reply src/core/types/chains.ts:42]: the new Chain object from viem is different than Chain object from wagmi 🙃

**Review by greg-schrammel — APPROVED** [review]

**Comment by DanielSinclair** [comment]
Seeing an issue when building locally; my ENV does include the new Blast RPCs. Switch network RPC call seems to work fine.
Screenshot 2024-03-07 at 6 02 10 PM
Screenshot 2024-03-07 at 6 02 20 PM

File: src/core/types/chains.ts (hunk 1 above)
Comment (jxom) [reply src/core/types/chains.ts:42]: Wat

Comment (estebanmino) [reply src/core/types/chains.ts:42]: https://github.com/rainbow-me/browser-extension/blob/master/package.json#L156

we're very late

**Comment by estebanmino** [comment]
> Seeing an issue when building locally; my ENV does include the new Blast RPCs. Switch network RPC call seems to work fine. Screenshot 2024-03-07 at 6 02 10 PM Screenshot 2024-03-07 at 6 02 20 PM

@DanielSinclair  ok it should be ok now

**Comment by DanielSinclair** [comment]
@estebanmino Tested again and seeing one last issue:
- Network appears as `Disabled` by default on a fresh install
Screenshot 2024-03-11 at 4 17 29 AM

**Review by BrodyHughes — APPROVED** [review]

**Pull request merged**